streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/CombinedKeySchema.java

```diff
@@ -94,7 +94,18 @@ public CombinedKey<KO, K> fromBytes(final Bytes data) {
         return new CombinedKey<>(foreignKey, primaryKey);
     }
 
-    Bytes prefixBytes(final KO key) {
+    public K deserializePrimaryKeyFromBytes(final Bytes data) {
```

Comment:
nice, this is a good optimization! it can be decoupled from the `prefixScan` part.

Reply:
What do you mean by decoupling here as it is already another method in another class?

Reply:
Oops, reading it back I can see why this is confusing. I just meant that we could get this change in separately from the `prefixScan` part if the backwards compatibility thing becomes a concern.

Reply:
As it was not public, which compatibility issue we could have if we remove or change this method? Also, if we keep the original implementation inside the default body of prefixScan for compatibility with user-defined stores, we can still use this new `deserializePrimaryKeyFromBytes` method

Reply:
I was referencing my other comment about compatibility of using `prefixScan` altogether, though since it seems like there's workarounds we can ignore this comment for now.